### DISCUSSION TITLE
Proper fix of time zone inconsistency in Altair or VegaLite?

### DISCUSSION THREAD

**Discussion by qnzhou** [vega/altair#2977] [Q&A]
It seems to be a known problem that Altair/VegaLite interprets dates like `2018-01-01` using UTC time zone, but visualizes the same data using local time zone (e.g. https://github.com/altair-viz/altair/issues/2077). The proposed solution is to either use Pandas which provides more consistent time zone handling or use `utc` prefix to visualize dates in UTC time zone.

However, the dataset I have contains more than 5000 rows, and I would like to load it directly from the URL so Altair does not dump the data together with Vega-Lite json spec. This constraint means I cannot use pandas to preprocess my data. Using `utc` prefix is viable, but I feel it makes the code confusing as the dates are really not in UTC... Is there any way of properly fixing the time zone problem directly within Altair or Vega-Lite? Thanks!

**Comment by jonmmease** [comment]
Hi @qnzhou, 
To work with the data in pandas and get around the 5,000 row limit you could look at using [VegaFusion](https://vegafusion.io/) to move the data aggregations to the Python kernel.

As for a pure Altair/Vega-Lite solution, the only option I can think of is to use a [calculate transform](https://altair-viz.github.io/user_guide/transform/calculate.html) to convert the date string (e.g. `2018-01-01`) to a timestamp string (e.g. `2018-01-01T00:00:00`) before the string is interpreted as a timestamp.

Something like:
```python
alt.Chart(url).transform_calculate(
    date="datum['date'] + 'T00:00:00'",
).mark_bar().encode(...)
```

Hope that helps!
